bob-common/src/data.rs

```diff
@@ -265,6 +265,38 @@ impl DiskPath {
         DiskPath { name, path }
     }
 
+    pub fn dev_name(&self) -> String {
+        let df = std::process::Command::new("df").arg(&self.path).output();
+        match df {
+            Ok(output) => {
+                if let (true, Ok(out)) = (output.status.success(), String::from_utf8(output.stdout))
+                {
+                    let mut lines = out.lines();
+                    lines.next(); // skip headers
+                    if let Some(line) = lines.next() {
+                        if let Some(raw_dev_name) = line.split_whitespace().next() {
+                            if let (Some(slash_ind), Some(non_digit_ind)) =
```

Comment:
Parsing code looks dangerous. Did you check, how this crate works: https://docs.rs/block-utils/latest/block_utils/fn.get_device_from_path.html ?

Reply:
This function returns info only when provided with path of format "/dev/<device_name>", so it can't be used to get device name. It could be possible to get device name via getting info about all devices on this computer (get_block_devices() & get_all_device_info() from this crate) and then comparing device_id of given disk path with all dev_id's of devices. But get_all_device_info() don't provide dev_id's of devices (returning None instead), at least on my computer.

Reply:
Ok. Thanks for checking